mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLHostOps.td

```diff
@@ -0,0 +1,42 @@
+//===-- SYCLHostOps.td - SYCL dialect operation definitions *- tablegen -*-===//
+//
+// MLIR-SYCL is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This is the operation definition file for SYCL dialect operations involving
+// host code.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SYCL_HOST_OPS
+#define SYCL_HOST_OPS
+
+include "mlir/Dialect/LLVMIR/LLVMOpBase.td"
+
+////////////////////////////////////////////////////////////////////////////////
+// BASE DECLARATIONS
+////////////////////////////////////////////////////////////////////////////////
+
+class SYCL_HostOp<string mnemonic, list<Trait> traits = []> :
+    SYCL_Op<"host." # mnemonic, traits>;
+
+////////////////////////////////////////////////////////////////////////////////
+// OPERATIONS
+////////////////////////////////////////////////////////////////////////////////
+
+def SYCLHostConstructorOp : SYCL_HostOp<"constructor"> {
+  let summary = [{
+    Operation representing the construction of a SYCL object, i.e., allocation,
+    member initialization and possible casts to the required address space.
```

Comment:
> and possible casts to the required address space.

We shouldn't have address spaces on the host side

Reply:
True. Updated.

Reply:
The PR description still mentions address spaces, so it would go into the commit message. Should probably fix that there, too.